k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/model/K8sManifests.kt

```diff
@@ -13,13 +13,34 @@ data class K8sData(
     val identity: String? = null
 )
 
+data class Condition(
+    val lastTransitionTime: String? = null,
+    val message: String? = null,
+    val reason: String? = null,
+    val status: String? = null,
+    val type: String? = null,
+)
+
+data class Status (
+    val conditions: Array<Condition>?,
+)
+
+inline fun Status.isReady() : Boolean {
+    this.conditions?.forEach { it ->
+        if (it.type == "Ready" || it.type == "Available")
+            return it.status == "True"
+    }
+    return true
+}
+
 open class K8sManifest(
-    open val apiVersion: String?,
-    open val kind: String?,
+    open var apiVersion: String?,
```

Comment:
Why do these need to be `var`?  Because of `jacksonObjectMapper` ?

Reply:
we rewrite values for `metadata`, `spec`, etc in the tests to check corner cases. so all attribtues except for `apiVersion` and `kind` were already variable attributes. I just changed these for consistency.